lib/jobs/dell-wsman-firmware-update-job.js

```diff
@@ -0,0 +1,204 @@
+                }
+                else {
+                    self.targetConfig.serverAddress = obm.config.host;
+                    self.targetConfig.userName = obm.config.user;
+                    self.targetConfig.password = encryption.decrypt(obm.config.password);
+                }
+            });
+    };
+
+    /*
+     * This method call the microservice to apply the firmware on target server
+     */
+    wsmanFirmwareUpdateJob.prototype._handleAsyncRequest = function(){
+        logger.info("calling to start firmware on target server" + this.targetConfig.serverAddress);
+        var self = this;
+        var apiHost = self.apiServer;
+        var path = self.firmwareConfigs.updater;
+        var method = 'POST';
+        var callBackUriRef = self.callbackUri;
+        var callbackIdentifier = uuid.v4();
+        callBackUriRef = callBackUriRef.replace(/_IDENTIFIER_/, callbackIdentifier);
+
+        self._subscribeHttpResponseUuid(self.firmwareUpdateCallback, callbackIdentifier);
+
+        var request = {
+            serverAddress: self.targetConfig.serverAddress,
+            userName: self.targetConfig.userName,
+            password: self.targetConfig.password,
+            shareAddress: self.shareConfig.shareAddress,
+            shareName: self.shareConfig.shareName,
+            catalogFileName: "Catalog.xml",
+            shareUserName: self.shareConfig.shareUserName,
+            sharePassword: self.shareConfig.sharePassword,
+            applyUpdate: "1",
+            mountPoint: "",
+            shareType: self.shareConfig.shareType,
+            rebootNeeded: self.rebootNeeded,
+            callBack: {
+                callbackUri: callBackUriRef,
+                callbackGraph: "Graph.Dell.Wsman.Update.Firmware",
+                type: "wsman"
+            }
+        };
+        logger.info('Before calling the endpoint :');
+        return self.clientRequest(apiHost, path, method, request);
+    };
+
+    /*
+     * function called after the result have been sent back from firmware update micro servcie
+     */
+
+    wsmanFirmwareUpdateJob.prototype.firmwareUpdateCallback = function firmwareUpdateCallback(data){
+        var self = this;
+        logger.info(' Fimware Update callback function invoked for Node: ' + self.nodeId + ' Type: ' + data.type); //jshint ignore:line
+        self.printResult(data);
+        var isSuccessful = false;
+        var failureFound = false;
+        /*
+         * callback data may have different result as Success, Exception, Failed. Only get
+         * all success result could be consider as success
+         */
+        if (data !== null && data.length > 0) {
+            data.forEach(function (item) {
+                if ( !(item.status === 'undefined' || item.status === null || item.status === ""|| failureFound === true)){
+                    if(item.status === 'Failed') {
+                        isSuccessful = false;
+                        failureFound = true;
```

Comment:
what's the difference with isSuccessful?

Reply:
If the array has multiple result field only isSussessful is insufficient to judge. Add a "failurlfound" flag just like a break operation(break could not be used in foreach).